src/data/session_playlists.py

```diff
@@ -63,10 +79,29 @@ def save_playlist(playlist):
     """
     existing_playlists = get_playlists()
     updated_playlists = [
-        playlist if playlist["id"] == existing_playlist["id"] else existing_playlist
+        {**existing_playlist, **playlist}
+        if playlist["id"] == existing_playlist["id"]
+        else existing_playlist
         for existing_playlist in existing_playlists
     ]
+    print(updated_playlists)
```

Comment:
Minor but one we'll talk about a bit more later - this `print` is really useful for debugging but could easily share personal data later (granted less likely than a todo, but still plausible it might say "Grandma Josephine's 80th Birthday Playlist" or something) - so worth thinking early in a project whether you might regret that, and have to lock down log access more heavily because of it

Reply:
Good question - maybe!

Having no print statements will hopefully make you confident that you're unlikely to leak sensitive data, but you might have a genuine want for some logging statements. So when we do introduce this later we'll start by switching to using e.g. `app.logger.info("this will be printed")` instead, which means we can start to say things like:
* Devs should use print statements for temporary logs, so that we can lint for print statements and block those getting merged
* Things we genuinely want logged can be reviewed in PRs, and pass the lint since they use a more formal logging approach

But really the focus here is on making sure the team has thought about this, and has _an approach_ to preventing sensitive info being logged. Which might just be allowing print, but looking out for it in PRs like this! 😄

Reply:
Yeah that makes a lot of sense. In my previous project we went down the route you mentioned. We used a logger for anything we needed logs for, which I suppose doesn't really protect you against logging sensitive info. But hopefully means that you think about it a bit more, since for local debugging it's more quick/reflexive to write `print` than `app.logger.info`.